op-chain-ops/genesis/check.go

```diff
@@ -64,24 +66,36 @@ func CheckPredeploys(db vm.StateDB) error {
 			return fmt.Errorf("no code found at predeploy %s", addr)
 		}
 
-		// There must be an implementation
-		impl := db.GetState(addr, ImplementationSlot)
-		implAddr := common.BytesToAddress(impl.Bytes())
-		if implAddr == (common.Address{}) {
-			return fmt.Errorf("no implementation for %s", addr)
-		}
-		implCode := db.GetCode(implAddr)
-		if len(implCode) == 0 {
-			return fmt.Errorf("no code found at predeploy impl %s", addr)
-		}
-
 		// There must be an admin
 		admin := db.GetState(addr, AdminSlot)
 		adminAddr := common.BytesToAddress(admin.Bytes())
-		if adminAddr != predeploys.ProxyAdminAddr {
-			return fmt.Errorf("admin is %s when it should be % for %s", adminAddr, predeploys.ProxyAdminAddr, addr)
+		if addr != predeploys.ProxyAdminAddr && addr != predeploys.GovernanceTokenAddr && adminAddr != predeploys.ProxyAdminAddr {
```

Comment:
Sanity check: is it correct to not also skip `LegacyERC20ETHAddr` here? What's the exact difference with the `special` cases in `mapImplementationAddress`?

Reply:
This loop only iterates over `0x42...0` - `0x42..0800`. The legacy ERC20 address is in `0xDead`, so won't be included.

Reply:
The `special` cases are those that aren't mapped to a proxy.